lib/java/com/google/android/material/button/MaterialButton.java

```diff
@@ -781,6 +782,17 @@ private void updateIcon(boolean needsIconReset) {
 
     if (hasIconChanged) {
       resetIconDrawable();
+      adjustTextFromIcon();
+    }
+  }
+
+  private void adjustTextFromIcon() {
+    if (icon != null) {
+      if (isIconViewStart()) {
+          setIconPadding(-icon.getIntrinsicWidth());
+      } else if (isIconViewEnd()) {
+        setIconPadding(icon.getIntrinsicWidth());
```

Comment:
Why do we need different logic here? Isn't the purpose "consuming" the icon width with a negative padding?